openpype/hosts/maya/plugins/publish/validate_rig_output_ids.py

```diff
@@ -30,43 +38,48 @@ def process(self, instance):
 
     @classmethod
     def get_invalid(cls, instance, compute=False):
-        invalid = cls.get_invalid_matches(instance, compute=compute)
-        return [x["node"].longName() for x in invalid]
+        invalid_matches = cls.get_invalid_matches(instance, compute=compute)
+        return list(invalid_matches.keys())
 
     @classmethod
     def get_invalid_matches(cls, instance, compute=False):
-        invalid = []
+        invalid = {}
 
         if compute:
             out_set = next(x for x in instance if x.endswith("out_SET"))
-            instance_nodes = pc.sets(out_set, query=True)
-            instance_nodes.extend(
-                [x.getShape() for x in instance_nodes if x.getShape()])
 
-            scene_nodes = pc.ls(type="transform") + pc.ls(type="mesh")
+            instance_nodes = cmds.sets(out_set, query=True, nodesOnly=True)
+            instance_nodes = cmds.ls(instance_nodes, long=True)
```

Comment:
does this actually return any extra instance_nodes?

Reply:
You mean the second `cmds.ls()` query? It enforces that the result is actually the full long name.